tqdm/std.py

```diff
@@ -127,6 +127,38 @@ def create_th_lock(cls):
 TqdmDefaultWriteLock.create_th_lock()
 
 
+class CallbackIOWrapper(object):
+    def __getattr__(self, name):
+        return getattr(self._wrapped, name)
+
+    def __setattr__(self, name, value):
+        return setattr(self._wrapped, name, value)
+
+    def __init__(self, callback, stream, method="read"):
+        """
+        Wrap a given `file`-like object's `read()` or `write()` to report
+        lengths to the given `callback`
+        """
+        object.__setattr__(self, '_wrapped', stream)
+        func = getattr(stream, method)
+        if method == "write":
+            @wraps(stream.write)
```

Comment:
I don't think you should use `@wraps` here. This is a separate file-like object with its own write method.

Reply:
given the `__get/setattr__` on the underlying `_wrapped` object I think it's fair to use `@wraps` though.